### PR TITLE
feat(doc): add autodoc_pydantic to sphinx

### PR DESCRIPTION
**Pull request by tcrasset**
Before:

image

After:

image

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -13,21 +13,26 @@ pydantic = "^1.9.0"
 pandas = "^1.4.2"
 toolz = "^0.12.0"
 
-[tool.poetry.group.dev.dependencies]
-blacken-docs = "^1.12.1"
-Sphinx = "^4.5.0"
+
+[tool.poetry.group.doc.dependencies]
```
Comment (tas17) [inline pyproject.toml:17]: Good point

**Review by tas17 — APPROVED** [review]
Nice :)

File: pyproject.toml (hunk 1 above)
Comment (jpetot) [reply pyproject.toml:17]: I don't see why we have to separate them

**Review by jpetot — APPROVED** [review]
👍

**Comment by tcrasset** [comment]
@jpetot We don't _have_ to, it's easier IMO to reason about all the dependencies if they are grouped in logical units

**Pull request merged**